docs/react-v9/contributing/rfcs/react-components/convergence/z-index-handling.md

```diff
@@ -0,0 +1,319 @@
+![z-index inconsistencies](assets/arbitrary-z-index-values.png)
+
+## Detailed Design or Proposal
+
+The creation of a global z-index system will help to standardize the values across the library, making it easier to manage and avoid conflicts between components. This system will also provide a way to easily override the z-index values for custom components and applications.
+
+## Research
+
+### CSS Isolation and stacking context
+
+[CSS Isolation](https://developer.mozilla.org/en-US/docs/Web/CSS/isolation) is a new feature that allows developers to create a new stacking context for an element. This means that the element and its children will be rendered isolated from the rest of the page. This can be useful explicitly for components that require a z-index control.
+The problem for our case is that we render most of our components that require a z axis control in a portal, which by itself creates a new stacking context. For the elements rendered in a portal, the z-index is relative to the portal element and we need a way to define the priority of them.
+
+## Layering/elevation concept
+
+As stated by [Fluent 2 design guidelines](https://fluent2.microsoft.design/elevation), objects/components have a defined elevation value to express its importance and create the sense of hierarchy of layers. These layers should have well defined shadow and z-axis values. Currently, the library have [shadows already defined](https://github.com/marcosmoura/fluentui/blob/943d1f166c5929ff65a683fbed3885434f16f2b9/packages/tokens/src/utils/shadows.ts#L6), but lacking the z-axis values.
+
+### What will be covered
+
+This proposal utilize this elevation system to define layers that translate into z-axis values, and those can be used by the components. The concepts are:
+
+- Each layer have a well-thought z-index number mapped to it
+- Multiple components can occupy the same layer, if they share the same importance level in the stack (e.g. Drawer and Nav)
+- A component can be defined without a z-index. Its placement fallbacks to its position within the [stack context](https://web.dev/learn/css/z-index/#stacking-context).
+
+### What won't be covered
+
+- Negative z-indexes. Components can define negative z-index values to represent background objects. Since this is very subjective, it'll be up to the component to define its own negative z-index values.
+- Correlation between shadows and z-indexes. Even though it might be related, this is out of the scope of this proposal. If necessary, it can be covered by a follow-up RFC.
+
+## Options
+
+### Issues
+
+The approaches below only standardize the layers and define z-index values for them. This is great for the current state of our components that can define arbitrary values, but won't solve a very specific problem: Defining priority for similar UI elements. e.g. Two Dropdowns are created. Which of them should have higher priority and therefore be displayed on top?
```

Comment:
```suggestion
> [!NOTE]  
> The approaches below only standardize the layers and define z-index values for them. This is great for the current state of our components that can define arbitrary values, but won't solve a very specific problem: Defining priority for similar UI elements. e.g. Two Dropdowns are created. Which of them should have higher priority and therefore be displayed on top?
```

Is it more like a note? Was a bit confused as it goes immediately as the first item in "Options" section